compiler/rustc_parse/src/parser/item.rs

```diff
@@ -291,6 +273,8 @@ impl<'a> Parser<'a> {
         } else if vis.kind.is_pub() && self.isnt_macro_invocation() {
             self.recover_missing_kw_before_item()?;
             return Ok(None);
+        } else if self.token.is_ident_named(sym::import) && self.isnt_macro_invocation() {
```

Comment:
I love this diagnostic improvement! Could we also consider doing the same thing for `using`? That keyword is used by C++ and C# for similar purposes (for instance: `using System.Collections.Generic;`). If you don't want to do that in this PR, that's totally fine! I can file an issue. 🙂

Reply:
Nah, that should be super easy.